network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/CalculatedBus.java

```diff
@@ -147,6 +147,54 @@ public Bus setAngle(double angle) {
         return this;
     }
 
+    @Override
+    public double getFictitiousP0() {
+        if (voltageLevelResource.getAttributes().getNodeToFictitiousP0() == null) {
+            return 0.0;
+        }
+        return voltageLevelResource.getAttributes().getNodeToFictitiousP0().values().stream().reduce(0.0, Double::sum);
+    }
+
+    @Override
+    public Bus setFictitiousP0(double p0) {
+        Map<Integer, Double> nodesMap = voltageLevelResource.getAttributes().getNodeToFictitiousP0();
+        // set the value in the first nodes map entry
+        int firstNode = 0;
+        if (nodesMap != null) {
+            nodesMap.replaceAll((k, v) -> 0.0);
```

Comment:
Here, you should set p0 to only the list of nodes corresponding to this calculated bus, no ?